src/Compilers/VisualBasic/Portable/Symbols/Source/SourceDelegateClonedParameterSymbolForBeginAndEndInvoke.vb

```diff
@@ -0,0 +1,39 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
+    Friend NotInheritable Class SourceDelegateClonedParameterSymbolForBeginAndEndInvoke
+        Inherits SourceClonedParameterSymbol
+
+        Public Sub New(originalParam As SourceParameterSymbol, newOwner As MethodSymbol, ordinal As Integer)
+            MyBase.New(originalParam, newOwner, ordinal)
+            Debug.Assert(Not originalParam.IsOptional)
+        End Sub
+
+        Friend Overrides ReadOnly Property IsCallerLineNumber As Boolean
+            Get
+                Return _originalParam.IsCallerLineNumber
+            End Get
+        End Property
+
+        Friend Overrides ReadOnly Property IsCallerMemberName As Boolean
+            Get
+                Return _originalParam.IsCallerMemberName
+            End Get
+        End Property
+
+        Friend Overrides ReadOnly Property IsCallerFilePath As Boolean
+            Get
+                Return _originalParam.IsCallerFilePath
+            End Get
+        End Property
+
+        Friend Overrides ReadOnly Property CallerArgumentExpressionParameterIndex As Integer
+            Get
+                ' This parameter cannot be optional. Hence, this property shouldn't be accessed.
```

Comment:
I'm unsure why this reasoning applies to this parameter, but not the rest of the caller info attributes?

Reply:
@333fred It should apply, but I avoided changing the existing implementation. Let me change it so things are less confusing.